### PR TITLE
feat: Add ci pipeline

### LINKED ISSUE #7: Feature: Add CI pipeline in GitHub Actions
**Is your feature request related to a problem? Please describe.**
A Continous Integration pipeline should run on this repo.

**Describe the solution you'd like**

- Prod build of all projects should be triggered during PR creation,
- Tests should be triggered,
- If build or testing failed, PR must be blocked,

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by KonradDyb**
## Summary

Add ci pipeline.

## Details

Add ci basic pipeline for future development. 

## References

_Links which provides to external materials_ <br/>
[Nice guide for create CI/CD pipeline](https://www.youtube.com/watch?v=R5ppadIsGbA) 
[GitHub guide](https://github.blog/2022-02-02-build-ci-cd-pipeline-github-actions-four-steps/)

### CODE REVIEW

File: .github/workflows/ci-cd.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Continous Integration and Deployment
```
Comment (StraykerPL) [inline .github/workflows/ci-cd.yaml]: For now it's just CI, no deploy is being performed.

File: .github/workflows/ci-cd.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+jobs:
```
Comment (StraykerPL) [inline .github/workflows/ci-cd.yaml]: Missing job for Web Frontend project prod build.

**Review by StraykerPL — CHANGES_REQUESTED** [review]

**Comment by StraykerPL** [comment]
Please update PR's data: Assignees, Labels and ticket connection.

File: .github/workflows/ci-cd.yaml (hunk 1 above)
Comment (KonradDyb) [reply .github/workflows/ci-cd.yaml]: Done.

File: .github/workflows/ci-cd.yaml (hunk 2 above)
Comment (KonradDyb) [reply .github/workflows/ci-cd.yaml]: Done.

**Comment by KonradDyb** [comment]
> Please update PR's data: Assignees, Labels and ticket connection.

Done.

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+    - name: Check out code 👀
+      uses: actions/checkout@v3
+
+    - name: Setup Node.js ${{ env.NODE_VERSION }} 🔨
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+        cache: 'npm'
+        cache-dependency-path: binder-web-frontend/package-lock.json
```
Comment (StraykerPL) [inline .github/workflows/pull-request.yml]: Is it possible to use `env` variable here too?

**Review by StraykerPL — CHANGES_REQUESTED** [review]

**Comment by StraykerPL** [comment]
Mark addressed comments conversations as resolved.

**Comment by StraykerPL** [comment]
Update branch as requested by PR control.

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (KonradDyb) [reply .github/workflows/pull-request.yml]: Yes it was possible :) Nice catch !

**Review by StraykerPL — APPROVED** [review]

**Pull request merged**